src/synthetix/StakingRewards.t.sol

```diff
@@ -270,16 +286,120 @@ contract StakingRewardsTest is Test {
         assertEq(rewards.rewardsDuration(), 70 days);
     }
 
-    function testRevertSetRewardsDurationOnActiveDistribution() public {
-        setupStakingToken(100 * WAD);
-        rewards.stake(100 * WAD);
+    function checkChangeRewardDurationOnActiveDistribution(uint256 newDuration) private {
+        uint256 totalStake = 100 * WAD;
+        uint256 totalReward = 123 * WAD;
+        uint256 initialDuration = 7 days;
+        uint256 initialRun = 1 days;
+        initFarm(totalStake, totalReward, initialDuration);
 
-        setupReward(100 * WAD);
+        skip(initialRun);
 
-        skip(1 days);
+        rewards.setRewardsDuration(newDuration);
 
-        vm.expectRevert("Previous rewards period must be complete before changing the duration for the new period");
-        rewards.setRewardsDuration(70 days);
+        assertEq(rewards.rewardPerTokenStored(), initialRun * (totalReward / initialDuration) * WAD / totalStake);
+        assertEq(rewards.lastUpdateTime(), block.timestamp);
+        assertEq(rewards.rewardRate(), (initialDuration - initialRun) * (totalReward / initialDuration) / newDuration);
+        assertEq(rewards.periodFinish(), block.timestamp + newDuration);
+        assertEq(rewards.rewardsDuration(), newDuration);
+
+        skip(newDuration);
+
+        rewards.exit();
+
+        uint256 rewardLeft = totalReward % initialDuration + (initialDuration - initialRun) * (totalReward / initialDuration) % newDuration; // dust lost due to rewardRate being rounded down
+        uint256 rewardPaid = totalReward - rewardLeft;
+        assertEq(rewardGem.balanceOf(address(this)), rewardPaid);
+        assertEq(rewardGem.balanceOf(address(rewards)), rewardLeft);
+        assertEq(gem.balanceOf(address(this)), totalStake);
+        assertEq(gem.balanceOf(address(rewards)), 0);
+    }
+
+    function testIncreaseRewardsDurationOnActiveDistribution() public {
+        checkChangeRewardDurationOnActiveDistribution(70 days);
+    }
+
+    function testDecreaseRewardsDurationOnActiveDistribution() public {
+        checkChangeRewardDurationOnActiveDistribution(5 days);
+    }
+
+    function testSetSameRewardsDurationOnActiveDistribution() public {
+        uint256 totalStake = 100 * WAD;
+        uint256 totalReward = 123 * WAD;
+        uint256 initialDuration = 5 days;
```

Comment:
It is the exact same thing for the test purpose, but as before it was 7 days, not sure if this change was on purpose or not.

Reply:
Good catch, was not intentional. Fixed.